lib/chai/utils/getPathInfo.js

```diff
@@ -61,13 +62,13 @@ module.exports = function getPathInfo(path, obj) {
  */
 
 function parsePath (path) {
-  var str = path.replace(/\[/g, '.[')
+  var str = path.replace(/([^\\])\[/g, '$1.[')
     , parts = str.match(/(\\\.|[^.]+?)+/g);
   return parts.map(function (value) {
-    var re = /\[(\d+)\]$/
+    var re = /^\[(\d+)\]$/
```

Comment:
This `^` ensures `test\\[1]` parsed as the property `test[1]`, not the index `1`